Kernel/Devices/PATAChannel.h

```diff
@@ -0,0 +1,76 @@
+//
+// Parallel ATA (PATA) controller driver
+//
+// This driver describes a logical PATA Channel. Each channel can connect up to 2
+// IDE Hard Disk Drives. The drives themselves can be either the master drive (hd0)
+// or the slave drive (hd1).
+//
+// More information about the ATA spec for PATA can be found here:
+//      ftp://ftp.seagate.com/acrobat/reference/111-1c.pdf
+//
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <Kernel/IRQHandler.h>
+#include <Kernel/Lock.h>
+#include <Kernel/PCI.h>
+#include <Kernel/VM/PhysicalAddress.h>
+#include <Kernel/VM/PhysicalPage.h>
+
+struct PhysicalRegionDescriptor {
+    PhysicalAddress offset;
+    u16 size { 0 };
+    u16 end_of_table { 0 };
+};
+
+class PATADiskDevice;
+class PATAChannel final : public IRQHandler
+{
+    friend class PATADiskDevice;
+    AK_MAKE_ETERNAL
+public:
+    enum class ChannelType : u8
```

Comment:
Shouldn't this enum be `Primary`/`Secondary` rather than `Master`/`Slave`?

Reply:
I was actually thinking about this before. Seagate's spec (albeit 1993) used the term `Master / Slave` but I think `Primary / Secondary` is more commonly seen on `BIOS` screens so I'll change it.